### ISSUE TITLE
MS Farmer not working

### ISSUE THREAD

**Issue by obes2137** [charlesbel/Microsoft-Rewards-Farmer#154]
I installed everything from requirements.txt, even the webdriver thingy, yet it's not working for me
![python3ss](https://user-images.githubusercontent.com/109519703/217913977-a2fd9e9e-7507-49ca-8a7e-487abc61327d.png)

**Comment by Stales89** [comment]
I too have the same issue.

**Comment by AlexyGalaxy** [comment]
I have uploaded an updated version of this scrip to my github. It allows for more quizes to be completed as bing rewards keeps updating. I also added timers and counters and im working on adding a logger and error logger.

If you would like to check it out my github link is: https://github.com/AlexyGalaxy/Microsoft-Rewards-Farmer

If you have any issues comment below and I can try to help you.

**Comment by obes2137** [comment]
@AlexyGalaxy so I tried your fork of the app. I got somewhere, but I still end up getting an error when I try to run it. Am I doing something wrong or wut
![obraz](https://user-images.githubusercontent.com/109519703/218182803-faacc7dd-7452-4361-be8e-4b55f5568116.png)

**Comment by AlexyGalaxy** [comment]
It sounds like you never got the script fully running because it is looking for a particular element that is used in you region. Are you outside of the US?

Try changing waitUntilVisible to the code below and let me know what happens.
```
waitUntilVisible(browser, By.XPATH, '//*[@id="navs"]/div/div/div/div/div[4]/div/button/span/i', 20)
```
Also disable headless mode by commenting out line 46 and 47 with #. Then watch the script open chrome and look to see where it is stuck. (When done testing make sure to delete the # to rentable headless mode.
```
#if headless_mode : #comment out to disable headless mode (makes window visable) 
           #options.add_argument("--headless") #comment out to disable headless mode (makes window visable) 
```
Pause the script by placing the code located below, before waitUntilVisible and once again after waitUntilVisible. Then run the script and post the screen shot here. If the browser is taking too long to load, increase the number at the end of waitUntilVisible to 30-50(this is the time waited)
```
print('[TEST] Press any key to continue')
input()
```
If it doesnt work read this:
The way this line of code works is by grabbing the XPATH information. When your script is paused and waiting for your second input to continue, click on the browsers that was opened and running the script, press F12(this will open a page showing the browsers code), then press ctrl+shift+c, then hover over the top banner of the website and right click. Now in that new page that shows the browsers code there is a line highlighted. Right click that line and select Copy>Copy XPATH. Paste that line of code within waitUntilVisible.
```
waitUntilVisible(browser, By.XPATH, 'PASTE_XPATH_CODE_HERE/DELETE_THESE_WORDS_WITHIN_THE_QUOTES', 20)
```
let me know if that helps. If so please commnet below with the changes you made to the script.

Someone had this exact issue but unfortunately they did not post the exact changes they made.
https://github.com/charlesbel/Microsoft-Rewards-Farmer/issues/152

**Comment by obes2137** [comment]
@AlexyGalaxy So, I did the first three things, however i'm getting an error. My guess would be that I did something wrong while following your instructions. And yes, i'm outside of the US
![obraz](https://user-images.githubusercontent.com/109519703/218278451-db382513-1b6e-43cd-a871-82f32697b2bb.png)
![obraz](https://user-images.githubusercontent.com/109519703/218278472-a353646b-9273-49bb-9f43-d38dda6dc7bb.png)

**Comment by AlexyGalaxy** [comment]
![img1](https://user-images.githubusercontent.com/86795638/218284726-f8e4f802-c7d5-4687-845b-40045eafe92f.JPG)
You need to hit TAB before input() so that everything is in line. With this coding language, python, you need to always keep code in line with the line before. Follow the syntax in my screenshot.

**Comment by obes2137** [comment]
![obraz](https://user-images.githubusercontent.com/109519703/218315456-80078db7-6c3b-4cd1-be8a-b383dfc6dea7.png)
![obraz](https://user-images.githubusercontent.com/109519703/218315500-7c28fd62-6d9c-40a7-bb25-1643fd6c5ee4.png)
@AlexyGalaxy

**Comment by AlexyGalaxy** [comment]
ive never seen that one before... try using space instead of tab and let me know if it fixes the error

**Comment by obes2137** [comment]
it wants me to correct everything with spaces now
![obraz](https://user-images.githubusercontent.com/109519703/218554682-f6a8d7b8-aea1-417c-90dd-a15d63c67304.png)

**Comment by AlexyGalaxy** [comment]
Thats very weird.... I am not sure what to say or do because when I google it people say "just delete the tabs" but I myself do not experience this error

**Comment by charlesbel** [comment]
Fixed with v3.0 #188
